spec/ruby/core/numeric/shared/step.rb

```diff
@@ -0,0 +1,397 @@
+    end
+
+    it "raises an ArgumentError if given a block" do
+      lambda { 1.send(@method, *get_args(@args_type, 5, "1")) {} }.should raise_error(ArgumentError)
+      lambda { 1.send(@method, *get_args(@args_type, 5, "0.1")) {} }.should raise_error(ArgumentError)
+      lambda { 1.send(@method, *get_args(@args_type, 5, "1/3")) {} }.should raise_error(ArgumentError)
+      lambda { 1.send(@method, *get_args(@args_type, 5, "foo")) {} }.should raise_error(ArgumentError)
+    end
+  end
+
+  describe "Numeric#step with [stop, step] when self and stop are Floats but step is a String" do
+    it "returns an Enumerator if not given a block" do
+      1.1.send(@method, *get_args(@args_type, 5.1, "1")).should be_an_instance_of(enumerator_class)
+      1.1.send(@method, *get_args(@args_type, 5.1, "0.1")).should be_an_instance_of(enumerator_class)
+      1.1.send(@method, *get_args(@args_type, 5.1, "1/3")).should be_an_instance_of(enumerator_class)
+      1.1.send(@method, *get_args(@args_type, 5.1, "foo")).should be_an_instance_of(enumerator_class)
+    end
+
+    it "raises a ArgumentError if given a block" do
+      lambda { 1.1.send(@method, *get_args(@args_type, 5.1, "1")) {} }.should raise_error(ArgumentError)
+      lambda { 1.1.send(@method, *get_args(@args_type, 5.1, "0.1")) {} }.should raise_error(ArgumentError)
+      lambda { 1.1.send(@method, *get_args(@args_type, 5.1, "1/3")) {} }.should raise_error(ArgumentError)
+      lambda { 1.1.send(@method, *get_args(@args_type, 5.1, "foo")) {} }.should raise_error(ArgumentError)
+    end
+  end
+
+  describe "Numeric#step with [stop, +step] when self, stop and step are Fixnums" do
+    it "yields while increasing self by step until stop is reached" do
+      1.send(@method, *get_args(@args_type, 5, 1), &@prc)
+      ScratchPad.recorded.should == [1, 2, 3, 4, 5]
+    end
+
+    it "yields once when self equals stop" do
+      1.send(@method, *get_args(@args_type, 1, 1), &@prc)
+      ScratchPad.recorded.should == [1]
+    end
+
+    it "does not yield when self is greater than stop" do
+      2.send(@method, *get_args(@args_type, 1, 1), &@prc)
+      ScratchPad.recorded.should == []
+    end
+  end
+
+  describe "Numeric#step with [stop, -step] when self, stop and step are Fixnums" do
+    it "yields while decreasing self by step until stop is reached" do
+      5.send(@method, *get_args(@args_type, 1, -1), &@prc)
+      ScratchPad.recorded.should == [5, 4, 3, 2, 1]
+    end
+
+    it "yields once when self equals stop" do
+      5.send(@method, *get_args(@args_type, 5, -1), &@prc)
+      ScratchPad.recorded.should == [5]
+    end
+
+    it "does not yield when self is less than stop" do
+      1.send(@method, *get_args(@args_type, 5, -1), &@prc)
+      ScratchPad.recorded.should == []
+    end
+  end
+
+  describe "Numeric#step with [stop, step]" do
+    it "yields only Floats when self is a Float" do
+      1.5.send(@method, *get_args(@args_type, 5, 1)) { |x| x.should be_kind_of(Float) }
```

Comment:
Use `should be_an_instance_of` instead as we want to test if the object is an actual instance.